### PR TITLE
Capture and persist all entity updates

### PR DESCRIPTION
**Pull request by patnorris**
Completes the initial version of the edit mode via customizing A-Frame's Inspector.
All edits made by the user in the Inspector are now persisted in the backend canister when the user clicks on the Save button.

### CODE REVIEW

File: src/PersonalWebSpace_frontend/store.ts [hunk 1]
```diff
@@ -10,20 +10,15 @@ import {
...
+/* export const HOST =
   process.env.DFX_NETWORK !== "ic"
     ? "http://localhost:4943" // to be used with http://localhost:4943/?canisterId=ryjl3-tyaaa-aaaaa-aaaba-cai#/testroom
-    : "https://ic0.app";
+    : "https://ic0.app"; // Use in Production (on Mainnet) */
 
-/* export const HOST =
-  process.env.DFX_NETWORK === "ic"
+export const HOST =
+  backendCanisterId === "vee64-zyaaa-aaaai-acpta-cai"
```
Comment (tiusty) [inline src/PersonalWebSpace_frontend/store.ts:19]: Can we import the alias rather then manual define it?

File: src/PersonalWebSpace_frontend/store.ts [hunk 2]
```diff
@@ -10,20 +10,15 @@ import {
...
     ? "https://ic0.app"
```
Comment (tiusty) [inline src/PersonalWebSpace_frontend/store.ts:20]: what is ic0.app url?

File: src/PersonalWebSpace_frontend/store.ts [hunk 3]
```diff
@@ -10,20 +10,15 @@ import {
 } from "../declarations/PersonalWebSpace_backend";
 
 //__________Local vs Mainnet Development____________
-export const HOST =
+/* export const HOST =
   process.env.DFX_NETWORK !== "ic"
```
Comment (tiusty) [inline src/PersonalWebSpace_frontend/store.ts:14]: Does this not let us check for mainet vs local host?

File: src/PersonalWebSpace_frontend/pages/Space.svelte [hunk 4]
```diff
@@ -118,27 +119,37 @@
...
-  const enableExportOfAllChangesMade = () => {
-    // Background: changes made in the Inspector's scene view are only reflected in the updated HTML (exported during save event) for properties that had already been present in the entity's initial HTML element
-      // These properties have their dedicated PropertyRow (in the right panel) marked with a second class, propertyRowDefined (in addition to the class propertyRow)
-      // Thus, all property rows in the right panel need to have the second class propertyRowDefined and we add it to those which don't have it yet (because the property isn't present in the entity's initial HTML element)
-    var elements = document.body.getElementsByClassName("propertyRow");
-    console.log('in enableExportOfAllChangesMade elements', elements);
-    if(elements && elements.length > 0) {
-      //[ ...elements ].forEach( x => x.className += ' btn' )    
-      [ ...elements ].forEach( x => x.classList.add('propertyRowDefined') );
+  const enableExportOfAllChangesMade = (entityUpdateEvent) => {
```
Comment (tiusty) [inline src/PersonalWebSpace_frontend/pages/Space.svelte:122]: In what cases do the updated inspector get used? I assume not on any generic Aframe file like testroom. This only gets fixed for cases when you are editing the room in the space.svelte page?

**Comment by tiusty** [comment]
I created an issue #4 that let me not render spaces/failed to load spaces. This may be related to this branch or maybe not? Therefore I can't load/edit spaces so hard for me to test whether this works or not for me

**Review by tiusty — APPROVED** [review]

File: README.md [hunk 5]
```diff
@@ -33,6 +33,7 @@ npm install
 https://github.com/dfinity/vessel:
 
 npm run dev
+Note: this starts a clean local IC replica (i.e. all canister state is erased). To keep the current canister state, run dfx start --background instead
```
[the remark this answers is not in the document]
Comment (patnorris) [reply README.md:36]: good point, I updated the scripts accordingly, so npm run dev now starts a replica with the previous state persisted. For a fresh start, run npm run erase-replica. The Readme should also be clearer now.

File: src/PersonalWebSpace_frontend/pages/Space.svelte (hunk 4 above)
Comment (patnorris) [reply src/PersonalWebSpace_frontend/pages/Space.svelte:122]: That's right, when we load a space through .../#/space/spaceid and the inspector there, we're customizing the inspector with these functions. We can do more customizations here as well. In the future, we could also fork the inspector repo and host our own instance with all our changes already in there.

File: src/PersonalWebSpace_frontend/store.ts (hunk 3 above)
Comment (patnorris) [reply src/PersonalWebSpace_frontend/store.ts:14]: It actually doesn't work but I found a way now to use another env var for this check (NODE_ENV)

File: src/PersonalWebSpace_frontend/store.ts (hunk 2 above)
Comment (patnorris) [reply src/PersonalWebSpace_frontend/store.ts:20]: that's the default URL of the Mainnet Internet Computer where all canisters can be called (e.g. https://vdfyi-uaaaa-aaaai-acptq-cai.ic0.app/)

File: src/PersonalWebSpace_frontend/store.ts (hunk 1 above)
Comment (patnorris) [reply src/PersonalWebSpace_frontend/store.ts:19]: the backendCanisterId actually imports the alias for the canister (another way would be process.env.PERSONALWEBSPACE_BACKEND_CANISTER_ID as defined in vite.config.ts as canisterDefinitions)

**Comment by patnorris** [comment]
Thanks, I think it'll be a good idea for me to include some notes on how to test the changes in future PRs

**Pull request merged**